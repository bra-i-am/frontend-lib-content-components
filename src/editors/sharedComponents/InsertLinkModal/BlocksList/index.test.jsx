import React from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import BlocksList from '.';

Enzyme.configure({ adapter: new Adapter() });

const mockBlocks = {
  block1: { id: 'block1', path: 'Block 1', type: 'section' },
  block2: { id: 'block2', path: 'Block 2', type: 'subsection' },
};

describe('BlocksList Component', () => {
  // eslint-disable-next-line react/prop-types
  const IntlProviderWrapper = ({ children }) => (
    <IntlProvider locale="en" messages={{}}>
      {children}
    </IntlProvider>
  );

  test('renders without crashing', () => {
    const wrapper = shallow(
      <IntlProviderWrapper>
        <BlocksList blocks={mockBlocks} onBlockSelected={() => {}} />
      </IntlProviderWrapper>,
    );
    expect(wrapper.exists()).toBeTruthy();
  });
});
