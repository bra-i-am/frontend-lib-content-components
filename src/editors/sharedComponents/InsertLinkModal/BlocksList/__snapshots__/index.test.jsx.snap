// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlocksList Component snapshot 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  fallbackOnEmptyString={true}
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  onWarn={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <BlocksList
    blocks={
      Object {
        "block-children-1": Object {
          "blockId": "edx_block-1",
          "displayName": "Block children 1",
          "id": "block-children-1",
          "legacyWebUrl": "http://localhost/legacy",
          "lmsWebUrl": "http://localhost/weburl",
          "studentViewUrl": "http://localhost/studentview",
          "type": "sequential",
        },
        "block-children-2": Object {
          "blockId": "edx_block-2",
          "displayName": "Block children 2",
          "id": "block-children-2",
          "legacyWebUrl": "http://localhost/legacy",
          "lmsWebUrl": "http://localhost/weburl",
          "studentViewUrl": "http://localhost/studentview",
          "type": "sequential",
        },
        "block-key": Object {
          "blockId": "edx_block-1",
          "children": Array [
            "block-children-1",
            "block-children-2",
          ],
          "displayName": "Any display name",
          "id": "block-key",
          "legacyWebUrl": "http://localhost/legacy",
          "lmsWebUrl": "http://localhost/weburl",
          "studentViewUrl": "http://localhost/studentview",
          "type": "character",
        },
      }
    }
    onBlockSelected={[Function]}
  />
</IntlProvider>
`;
